Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -384,6 +388,10 @@ Ref<MediaTimePromise> MediaSource::waitForTarget(const SeekTarget& target)
     // 2.4.3 Seeking
     // https://rawgit.com/w3c/media-source/45627646344eea0170dd1cbc5a3d508ca751abb8/media-source-respec.html#mediasource-seeking
 
+    RefPtr msp = protectedPrivate();
+    if (!msp)
+        return MediaTimePromise::createAndReject(PlatformMediaError::SourceRemoved);
```

Comment:
This is new, is this needed, or should we use releaseNonNull()?

Reply:
it is new, and while I don't think it could happen, I wanted to cover the case.